doc/developers/maintainer.rst

```diff
@@ -107,34 +107,62 @@ under the ``doc/whats_new/`` folder so PRs that target the next version can
 contribute their changelog entries to this file in parallel to the release
 process.
 
-Minor version release
-~~~~~~~~~~~~~~~~~~~~~
+Minor version release and bug-fix release
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 The minor releases should include bug fixes and some relevant documentation
 changes only. Any PR resulting in a behavior change which is not a bug fix
 should be excluded.
 
-First, create a branch, **on your own fork** (to release e.g. `0.99.3`):
+ - Create a branch, **on your own fork** (here referred to as `fork`) for the release.
+   For instance, to release `1.2.2` create the `release-1.2.2` branch
+   from `main` with:
+
+    .. prompt:: bash $
+
+        git checkout main
+        git pull upstream main:main
+        git checkout -b release-1.2.2 main
+        git push -u fork release-1.2.2:release-1.2.2
+
+ - Create a PR to the `upstream/1.2.X` branch (not to `upstream/main`)
+   with all the desired changes.
+
+ - Do not push anything on that branch yet.
+
+ - Locally rebase `release-1.2.2` from the `upstream/1.2.X` branch using:
+
+    .. prompt:: bash $
+
+        git rebase -i upstream/1.2.X
+
+   This will open an interactive rebase with the `git-rebase-todo` containing all
+   the latest commit on `main`. At this stage, you have to perform
+   this interactive rebase with at least someone else (being three people rebasing
+   is better not to forget something and to avoid any doubt).
+
+     - **Do not remove lines but drop commit by replace** ``pick`` **with** ``drop``
 
-.. prompt:: bash $
+     - Commits to pick for bug-fix release *generally* are prefixed with: `FIX`, `CI`,
+       `DOC`
```

Comment:
```suggestion
       `DOC`. They should at least include all the commits of the merged PRs
       that were milestoned for this PR on github and/or documented as such in
       the changelog. It's likely that the some bugfixes where document in the
       changelog of the main major release instead of the next bugfix release,
       in which case, the matching changelog entries will need to be moved,
       first in the `main` branch then backported in the release PR.
```